### PR TITLE
Migrate to Python 3.6+

### LINKED ISSUE #289: Requirement is pinned for pyicloud 0.9.7: click>=6.0,<=7.1.1
## The problem

- This library supports an integration in Home Assistant.
- The upcoming pip resolver will be more strict in resolving package version conflicts:
http://pyfound.blogspot.com/2020/03/new-pip-resolver-to-roll-out-this-year.html#what-will-change
- To avoid future issues it would be good if this library would change its pinning to allow any higher version and just set a minimum version.

Thanks for your support of Home Assistant!

## Environment

- pyiCloud release with the issue (`pip show pyicloud`): 0.9.7
- Last working pyiCloud release (if known): 
- Service causing this issue: 
- Python version (`python -V`): 3.7.7
- Operating environment (project deps/Docker/Windows/etc.):  Ubuntu 18.04

## Traceback/Error logs

```shell
(home-assistant) x:~/Dev/home-assistant/home-assistant$ pip install -r requirements_all.txt
...
(home-assistant) x:~/Dev/home-assistant/home-assistant$ pipdeptree --reverse --packages click
...
click==7.1.1
...
  - pyicloud==0.9.7 [requires: click>=6.0,<=7.1.1]
...
```

### LINKED ISSUE #290: Requirement is pinned for pyicloud 0.9.7: keyring>=8.0,<=9.3.1 and keyrings.alt>=1.0,<=3.2.0
## The problem

- This library supports an integration in Home Assistant.
- The upcoming pip resolver will be more strict in resolving package version conflicts:
http://pyfound.blogspot.com/2020/03/new-pip-resolver-to-roll-out-this-year.html#what-will-change
- To avoid future issues it would be good if this library would change its pinning to allow any higher version and just set a minimum version.

Thanks for your support of Home Assistant!

## Environment

- pyiCloud release with the issue (`pip show pyicloud`): 0.9.7
- Last working pyiCloud release (if known): 
- Service causing this issue: 
- Python version (`python -V`): 3.7.7
- Operating environment (project deps/Docker/Windows/etc.):  Ubuntu 18.04

## Traceback/Error logs

```shell
(home-assistant) x:~/Dev/home-assistant/home-assistant$ pip install -r requirements_all.txt
...
(home-assistant) x:~/Dev/home-assistant/home-assistant$ pipdeptree --reverse --packages keyring
...
keyring==21.2.0
  - pyicloud==0.9.7 [requires: keyring>=8.0,<=9.3.1]
(home-assistant) x:~/Dev/home-assistant/home-assistant$ pipdeptree --reverse --packages keyrings.alt
...
keyrings.alt==3.4.0
  - pyicloud==0.9.7 [requires: keyrings.alt>=1.0,<=3.2.0]
```

### PR DESCRIPTION
**Pull request by Quentame**
Drop support of Python < 3.6
- remove six + future deps + usage
- update deps
- remove encoding
- remove (object)

Fixes #290 & fixes #289

### CODE REVIEW

File: pyicloud/base.py [hunk 1]
```diff
@@ -24,7 +23,6 @@
     RemindersService,
     PhotosService,
     AccountService,
-    DriveService,
```
Comment (Quentame) [inline pyicloud/base.py:27]: Don't remove this

File: pyicloud/base.py [hunk 2]
```diff
@@ -93,21 +91,20 @@ def request(self, method, url, **kwargs):  # pylint: disable=arguments-differ
...
+        code = data.get("errorCode")
+        if not code and data.get("serverErrorCode"):
+            code = data.get("serverErrorCode")
 
-                if reason:
-                    self._raise_error(code, reason)
+        if reason:
+            self._raise_error(code, reason)
 
         return response
```
Comment (Quentame) [inline pyicloud/base.py:109]: revert this

File: pyicloud/services/__init__.py [hunk 3]
```diff
@@ -6,4 +6,3 @@
 from pyicloud.services.reminders import RemindersService
 from pyicloud.services.photos import PhotosService
 from pyicloud.services.account import AccountService
-from pyicloud.services.drive import DriveService
```
Comment (Quentame) [inline pyicloud/services/__init__.py:9]: revert

File: pyicloud/services/account.py [hunk 4]
```diff
@@ -1,369 +1,44 @@
...
-
-    @property
-    def quota_paid(self):
-        """Gets the paid quota."""
-        return self.quota_data["paidQuota"]
-
-    def __unicode__(self):
-        return "%s%% used of %s bytes" % (
-            self.used_storage_in_percent,
-            self.total_storage_in_bytes,
```
Comment (Quentame) [inline pyicloud/services/account.py:332]: revert

File: pyicloud/services/calendar.py [hunk 5]
```diff
@@ -17,7 +17,7 @@ def __init__(self, service_root, session, params):
         self._service_root = service_root
         self._calendar_endpoint = "%s/ca" % self._service_root
         self._calendar_refresh_url = "%s/events" % self._calendar_endpoint
-        self._calendar_event_detail_url = "%s/eventdetail" % self._calendar_endpoint
+        self._calendar_event_detail_url = "%s/eventdetail" % (self._calendar_endpoint,)
         self._calendars = "%s/startup" % self._calendar_endpoint
```
Comment (Quentame) [inline pyicloud/services/calendar.py:21]: f-string

**Review by Quentame — COMMENTED** [review]
Check use of u-string from Py2 unicode strings
Use of f-string instead of concat

**Comment by RichieB2B** [comment]
Can this be updated and merged please? I just found out that pyicloud is the reason my python3 keyring is stuck at version 9.3.1 from 2016. :-(

**Comment by WhyNotHugo** [comment]
Hint: Lots of syntax can be automatically (and safely) upgraded by running:

    pyupgrade --py36-plus $(find . -name "*.py")

This'll also do things like upgrade string formatting syntax, and alike.

**Comment by WhyNotHugo** [comment]
Is this PR stale? I don't mind picking it up myself if you don't have the time.

**Comment by RichieB2B** [comment]
Go for it!

**Comment by WhyNotHugo** [comment]
I've opened #325.

**Pull request closed without merging**